pkg/services/ngalert/store/alert_rule.go

```diff
@@ -455,9 +472,28 @@ func (st DBstore) GetAlertRulesForScheduling(ctx context.Context, query *ngmodel
 			alertRulesSql += " WHERE " + filter
 		}
 
-		if err := sess.SQL(alertRulesSql, args...).Find(&rules); err != nil {
+		var faultyRules int64 = 0
+		rule := new(ngmodels.AlertRule)
+		rows, err := sess.SQL(alertRulesSql, args...).Rows(rule)
+		if err != nil {
 			return fmt.Errorf("failed to fetch alert rules: %w", err)
 		}
+		defer rows.Close()
+
+		for rows.Next() {
```

Comment:
WDYT about extracting the iteration + logging into a func to keep things a little more DRY? The caller could look more like:

```
alertRules := make([]*ngmodels.AlertRule, 0)
rows, err := sess.SQL(...).Rows(ngmodels.AlertRule{})
if err != nil {
    ...
}
go func() {
    _ = rows.Close()
}()
h.scanRules(rows, &alertRules)
```
Not a requirement, just a thought.